packages/vanilla/src/components/select/select.stories.js

```diff
@@ -0,0 +1,106 @@
+export default {
+  title: 'Patterns',
+  parameters: {
+    html: {
+      root: '.cbp-form'
+    }
+  },
+  argTypes: {
+    label: {
+      name: 'Label Element',
+      description: 'Represents a caption for the `<select>` element in the user interface.',
+      control: { type: 'text' }
+    },
+    inputDescription: {
+      name: 'Input Description',
+      description: 'Instructions or supplementary information regarding the `<select>` element.',
+      control: { type: 'text' }
+    },
+    labelFor: {
+      name: 'Label `for` Attribute',
+      description: 'Indicates the form element that this `<label>` describes and has the value which is the `id` of the `<select>` element it relates to.',
+      control: { type: 'text' }
+    },
+    formControlName: {
+      name: 'Input `name` Attribute',
+      description: 'Name of the form control in the input element. Submitted with the form as part of a name/value pair.',
+      control: { type: 'text' }
+    },
+    placeholderOption: {
+      name: 'Placeholder Option Text',
+      description: 'Text of the placeholder option.',
+      control: { type: 'text' }
+    },
+    errorMessage: {
+      name: 'Error Message',
+      description: 'A message in the input description that a problem has occurred.',
+      control: { type: 'text' }
+    },
+    optionsObj: {
+      name: 'Options Object',
+      description: 'Sets the label of the `<option>` and the `value` attribute.',
+      control: { type: 'object' }
+    },
+    disabled: {
+      name: '`disabled` Attribute',
+      description: 'Attribute that indicates that the user cannot interact with the control.',
+      control: { type: 'boolean' }
+    }
+  },
+  decorators: [
+    (Story, context) => `
+      <form class="cbp-form">
+        <label for=${context.args.labelFor} class="cbp-input__label">${context.args.label}</label>
+        <p class="cbp-input__description" ${!context.args.required ? '' : 'hidden'}>${context.args.inputDescription}</p>
```

Comment:
I don't see a `required` control that would set this. See my higher-level comment about this though.

Reply:
Forgot this control, should be added now along with the `aria-required` attribute. Toggling this control will show the css of the `:invalid` pseudo-class but regarding your comment below, we should revisit/discuss and finalize how to implement this along with the error message class.

Reply:
Yeah, I'm not sure why it would show the `:invalid` styling. Is this because you're still using the `required` attribute as well? This is certainly not desired behavior on an unfilled form and would be another reason not to use `required`.